src/panels/design/list-shmetro.tsx

```diff
@@ -27,30 +27,87 @@ const DesignListShmetro = () => {
 
 export default DesignListShmetro;
 
+const useStyles = makeStyles(theme =>
+    createStyles({
+        dividerVertical: {
+            margin: theme.spacing(0, 2),
+        },
+        nestedList: {
+            paddingLeft: theme.spacing(5),
+        },
+    })
+);
+
+const StationNumberSHMetroLi = () => {
+    const { t } = useTranslation();
+    const classes = useStyles();
+    const dispatch = useAppDispatch();
+
+    const station_number = useAppSelector(store => store.param.showStationNumber);
+    const line_number = useAppSelector(store => store.param.line_num);
+
+    return useMemo(() => {
+        const handleSwitch = (_: React.ChangeEvent<HTMLInputElement>, checked: boolean) => {
+            if (checked) {
+                dispatch(setShowStationNumber(true));
+            } else {
+                dispatch(setShowStationNumber(false));
+            }
+        };
+
+        const handleChange = ({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            dispatch(setLineNum(value));
+        };
+
+        return (
+            <>
+                <ListItem>
+                    <ListItemIcon>
+                        <Icon>account_balance_wallet</Icon>
+                    </ListItemIcon>
+                    <ListItemText primary={t('design.stationNumber')} />
```

Comment:
I believe users will be misinformed by the UI text. When you have a text indicating `Station Number`, they may change the input field below to see the update in station number. However, the input text is used to change the line number instead of the station number. Using the existing `<Line Number />` component in gzmtr might work. You may display this component only if `showStationNumber === true`.